File: frontend/mobile/google/android/MyHomeAgent/app/src/main/java/com/hestami_ai/myhomeagent/data/model/ServiceProvider.kt
package com.hestami_ai.myhomeagent.data.model

import com.google.gson.annotations.SerializedName

/**
 * Service provider model matching iOS ServiceProvider.swift implementation.
 */
data class ServiceProvider(
    @SerializedName("id")
    val id: String = "",

    @SerializedName("businessName")
    val businessName: String = "",

    @SerializedName("description")
    val description: String? = null,

    @SerializedName("rating")
    val rating: String = "",

    @SerializedName("totalReviews")
    val totalReviews: Int = 0
)

File: frontend/mobile/apple/iOS/My Home Agent/New Group/My Home Agent/Models/ServiceProvider.swift
import Foundation

struct ServiceProvider: Codable {
    let id: String
    let businessName: String
    let description: String?
    let rating: String
    let totalReviews: Int
    
    enum CodingKeys: String, CodingKey {
        case id
        case businessName
        case description
        case rating
        case totalReviews
    }
}